### PR TITLE
Don't pretend we support SharedArrayBuffer.

### LINKED ISSUE #26874: Hide SharedArrayBuffer
Right now we expose the SharedArrayBuffer interface through SpiderMonkey, but we have no code that supports passing it to DOM APIs or transferring it between threads. We should probably hide it instead or exposing something that isn't actually useful.

### PR DESCRIPTION
**Pull request by jdm**
Fixes #26874.

### CODE REVIEW

**Comment by gterzian** [comment]
Perhaps it's a good idea to also remove https://github.com/servo/servo/blob/33e96e95677c5c1a5d49fece4016b7acf1d7594b/components/script/script_runtime.rs#L460

So that we have a single commit turning off everything(making it easier to know what is required to turn it back on), and also maybe leaving the "can block" stuff while turning of SAB can lead to weirdness.

File: tests/wpt/metadata/xhr/send-data-sharedarraybuffer.any.js.ini [hunk 1]
```diff
@@ -1,5 +1,47 @@
...
+  [sending a Int32Array backed by a SharedArrayBuffer]
+    expected: FAIL
+  [sending a Uint32Array backed by a SharedArrayBuffer]
+    expected: FAIL
+  [sending a Float32Array backed by a SharedArrayBuffer]
+    expected: FAIL
+  [sending a Float64Array backed by a SharedArrayBuffer]
+    expected: FAIL
+  [sending a DataView backed by a SharedArrayBuffer]
+    expected: FAIL
```
Comment (gterzian) [inline tests/wpt/metadata/xhr/send-data-sharedarraybuffer.any.js.ini:47]: nit: do we always want to have a newline at the end of `.ini` files? If only to not have github show the warning in the diff?

**Review by gterzian — APPROVED** [review]
thanks for updating all those test expectations!

File: tests/wpt/metadata/xhr/send-data-sharedarraybuffer.any.js.ini (hunk 1 above)
Comment (jdm) [reply tests/wpt/metadata/xhr/send-data-sharedarraybuffer.any.js.ini:47]: I'm not convinced it's a useful thing to care about.

Comment (gterzian) [reply tests/wpt/metadata/xhr/send-data-sharedarraybuffer.any.js.ini:47]: ok!

**Pull request merged**